web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,888 @@
+                    checked: !checked,
+                },
+            }).then(function (value) {
+                self._setValue(value);
+            });
+        },
+        /**
+         * Method called when the wysiwyg instance is loaded.
+         *
+         * @private
+         */
+        _onLoadWysiwyg: function () {
+            var $button = this._renderTranslateButton();
+            $button.css({
+                'font-size': '15px',
+                position: 'absolute',
+                right: '+5px',
+                top: '+5px',
+            });
+            this.$el.append($button);
+        },
+        /**
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onWysiwygBlur: function (ev) {
+            ev.stopPropagation();
+            this._doAction();
+            if (ev.data.key === 'TAB') {
+                this.trigger_up('navigation_move', {
+                    direction: ev.data.shiftKey ? 'left' : 'right',
+                });
+            }
+        },
+        /**
+         * @private
+         * @param {OdooEvent} ev
+         */
+        _onWysiwygFocus: function (ev) {
+        },
+        _html_is_empty: function() {
+            //this will return us the latest content even if the widget is dirty and the 
+            //debounceAction is not done
+            //VERY IMPORTANT: LATEST REAL VALUE IF WIDGET isDIrty
+            var current_value = this.wysiwyg.getValue()
+            //LATEST SAVED VALUE OF THE SAME WIDGET 
+            //this._getValue()
+            //.isDirty will give us:
+            // this.wysiwyg.getValue() != this._getValue()
+            // We will not trigger .save on every onCHange for performance AND 
+            // functionality reasons.
+            var cleaned_content = current_value
+                .replace(/\s/g,'')
+                .replace(/\<p\>/g,'')
+                .replace(/\<\/p\>/g,'')
+                .replace(/\<\/br\>/g,'')
+                .replace(/\<br\>/g,'')
+                .replace(/&nbsp;/g,'')
+
+            return cleaned_content == ""
+            // Cleaned_content = cleaned_content.replace(/\<meta charset=\"utf-8\"\>/g,'') ;
+        },
+        _html_has_md_source_tag: function (){
+            var tag = this.$content.find('.web_widget_html_markdown_source');
+            return Boolean(tag.length)
+
+        },
+        _switch_markdown_html: function(ev) {
+            self = ev.data[0]
+            // If content empty, no warning needed.
+            // if tag is there, no warning needed.
+            if (ev.currentTarget.id == "markdown") {
+              var is_empty = self._html_is_empty();
+              var has_md_source_tag = self._html_has_md_source_tag();
+              if (!is_empty && !has_md_source_tag ) {
+                  // THIS SHOULD NEVER APPEAR, WE HAVE HIDDEN THE RADIOBUTTON IN ALL
+                  // POSSIBLE CIRCUMSTANCES, KEEPING IT FOR TESTING PURPOSES.
```

Comment:
Are we now keeping the radiobutton or not?

Reply:
Gio says we can clean it up.

Reply:
Done